src/get-started/java-binding.md

```diff
@@ -0,0 +1,382 @@
+          -DgroupId=com.example.myservice \
+          -DartifactId=my-service \
+          -Dversion=1.0
+```
+
+You can also use the interactive mode:
+
+``` none
+$ mvn archetype:generate \
+          -DarchetypeGroupId=com.exonum.binding \
+          -DarchetypeArtifactId=exonum-java-binding-service-archetype
+```
+
+The assembly description files for other systems can be created similarly to the
+template. For more information see an [example][assembly-description].
+
+#### System Dependencies
+
+System dependencies, such as `exonum-java-binding-core` and
+`exonum-java-binding-proofs`, are shipped together with the Exonum App, so when
+defining them in [pom.xml][pom.xml] you should use scope [`provided`](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope).
+
+### Service Development
+
+The service abstraction serves to expand the framework and implement the
+business logic of the application. The service defines the schema of the stored
+data that constitute the service state; transaction processing rules that can
+make changes to the stored data; and the external API that allows to interact
+with the service from outside of the system. See more information on the
+software model of the service in the [corresponding section][Exonum-services].
+
+In Java, the abstraction of the service is represented by
+`com.exonum.binding.service.Service` interface. Implementations can use the
+abstract class `com.exonum.binding.service.AbstractService`.
+
+### Service Schema Description
+
+Exonum provides a lot of stored named collections for the service data. The main
+collection types are sets, lists and maps. Data organization inside the
+collections is arranged in two ways - ordinary collections and [Merkelized
+collections](https://en.wikipedia.org/wiki/Merkle_tree); the latter allow to
+provide cryptographic evidence of the authenticity of the data to the clients of
+the system (for example, that an element is stored in the collection under a
+certain key). The [blockchain state](../glossary.md#blockchain-state) is
+influenced only by the Merkelized collections.
+
+For the detailed description of all Exonum collection types see the relevant
+[documentation](../architecture/storage/#table-types). In Java,
+implementations of the collections are located in
+`com.exonum.binding.storage.indices` package. This package documentation
+describes the peculiarities of their use.
+
+!!! note
+    `SparseListIndex` is not yet supported in Java. Let us know if it may be
+    useful for you!
+
+Collections work with a database view - either `Snapshot`, which is read-only;
+or `Fork`, which is mutable and allows performing modifying operations. The
+database representation is provided by the framework: `Snapshot` can be
+requested at any time, while `Fork` - only when the transaction is executed. The
+lifetime of these objects is limited by the scope of the method to which they
```

Comment:
Do I understand correctly that this lifetime restriction is true only for methods invoked by the core? That is, the `updateBalance` method below doesn't actually restrict the lifetime of its argument, right?

Reply:
That is correct, but the only way to access an instance of a view _in production code_ is to have core call you (either Transaction#execute or the lambda passed to Node#withSnapshot).

Shall we additionally clarify that?